packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -1087,13 +995,16 @@ describe("BnsConnection", () => {
       // this will never have tokens, but can try to sign
       const brokeIdentity = await profile.createIdentity(mainWalletId, chainId, HdPaths.simpleAddress(1234));
 
-      const sendTx: SendTransaction = {
+      const sendTx = (await connection.withDefaultFee({
         kind: "bcp/send",
         creator: brokeIdentity,
         recipient: await randomBnsAddress(),
         memo: "Sending from empty",
         amount: defaultAmount,
-      };
+      })) as SendTransaction;
```

Comment:
okay here we need `sendTx` with fees for the code below.

What about

```ts
const sendTx: SendTransaction = await connection.withDefaultFee({
  kind: "bcp/send",
  creator: brokeIdentity,
  recipient: await randomBnsAddress(),
  memo: "Sending from empty",
  amount: defaultAmount,
});
```

Reply:
That doesn't work (my first try), as the input expands the type from `kind: "bcp/send"` to `kind: string` when passed through the generic withDefaultFee. Thus we need an explicit cast. Explicit casts to other types do not work however, as they have the wrong fields (eg. amount and memo).

Try if you want, but this was the best I could do with the typings